src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -703,6 +703,37 @@ - (nullable NSData *)computePaseVerifier:(uint32_t)setupPincode iterations:(uint
     return result;
 }
 
+- (nullable NSData *)generateAttestationChallengeForDeviceId:(uint64_t)deviceId
+{
+    __block CHIP_ERROR errorCode = CHIP_ERROR_INCORRECT_STATE;
+    if (![self isRunning]) {
+        [self checkForError:errorCode logMsg:kErrorNotRunning error:nil];
+        return nil;
+    }
+
+    __block NSData * attestationChallenge;
+    dispatch_sync(_chipWorkQueue, ^{
+        if ([self isRunning]) {
+            chip::CommissioneeDeviceProxy * deviceProxy;
+            errorCode = self.cppCommissioner->GetDeviceBeingCommissioned(deviceId, &deviceProxy);
+            if (errorCode != CHIP_NO_ERROR) {
+                [self checkForError:errorCode logMsg:@"Invalid Attestation Challenge device ID." error:nil];
```

Comment:
The point of `checkForError` is that it returns whether there was an error, so can be:
```
   if ([self checkForError:errorCode ...]) {
       return;
   }
```
etc.